### PR TITLE
Add test case for parsing static value with parseJSONValue func

### PR DESCRIPTION
**Pull request by XiShanYongYe-Chang**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Add a new test case for parsing static value with parseJSONValue func, refer to https://github.com/karmada-io/karmada/issues/5788#issuecomment-2545862772

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/controllers/applicationfailover/common_test.go [hunk 1]
```diff
@@ -226,6 +226,15 @@ func Test_parseJSONValue(t *testing.T) {
 			wantErr: assert.NoError,
 			want:    "2",
 		},
+		{
```
Comment (zhzhuang-zju) [inline pkg/controllers/applicationfailover/common_test.go:229]: It would be nice to add a comment describing what the test case does.

**Review by zhzhuang-zju — COMMENTED** [review]
thanks, others LGTM

File: pkg/controllers/applicationfailover/common_test.go (hunk 1 above)
Comment (XiShanYongYe-Chang) [reply pkg/controllers/applicationfailover/common_test.go:229]: We can get it from the test case `get a static value`.

Comment (XiShanYongYe-Chang) [reply pkg/controllers/applicationfailover/common_test.go:229]: Update with `get a static value just as jsonPath write`

**Review by RainbowMango — COMMENTED** [review]
/hold
This behavior looks weird and still under discussion. Let's hold for a while.
See https://github.com/karmada-io/karmada/issues/5788#issuecomment-2548420154.

**Comment by XiShanYongYe-Chang** [comment]
From a usage perspective, this does not seem to be our recommended usage. Close it first.
/close

**Pull request closed without merging**